contracts/protocol/bases/OfferBase.sol

```diff
@@ -46,28 +53,44 @@ contract OfferBase is ProtocolBase, IBosonOfferEvents {
      * @param _offer - the fully populated struct with offer id set to 0x0 and voided set to false
      * @param _offerDates - the fully populated offer dates struct
      * @param _offerDurations - the fully populated offer durations struct
-     * @param _disputeResolverId - the id of chosen dispute resolver (can be 0)
+     * @param _drParameters - the id of chosen dispute resolver (can be 0) and mutualizer address (0 for self-mutualization)
      * @param _agentId - the id of agent
      * @param _feeLimit - the maximum fee that seller is willing to pay per exchange (for static offers)
      */
     function createOfferInternal(
         Offer memory _offer,
         OfferDates calldata _offerDates,
         OfferDurations calldata _offerDurations,
-        uint256 _disputeResolverId,
+        BosonTypes.DRParameters calldata _drParameters,
         uint256 _agentId,
         uint256 _feeLimit
     ) internal {
-        // get seller id, make sure it exists and store it to incoming struct
-        (bool exists, uint256 sellerId) = getSellerIdByAssistant(_msgSender());
-        if (!exists) revert NotAssistant();
-        _offer.sellerId = sellerId;
+        address sender = _msgSender();
+
+        if (_offer.creator == OfferCreator.Seller) {
```

Comment:
let's check here that when the seller is creating the offer, the `_offer.buyerId == 0`